### PR TITLE
Try to encapsulate retry logic in a clean way

### PR DESCRIPTION
**Pull request by dannymandel**
Hopefully this will make our OpenContext fetching run smoothly…

### CODE REVIEW

**Comment by dannymandel** [comment]
A variant of this is currently running on henry.cyverse.org 🤞

File: isb_lib/utilities/requests_utilities.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+L = logging.getLogger("requests_utilities")
+
+
+class RetryingRequests:
+    def __init__(self, include_random_on_failure: bool = False, timeout=60, max_retries=10):
+        self._include_random_on_failure = include_random_on_failure
+        self._timeout = timeout
+        self._max_retries = max_retries
+
+    def get(self, url: str, params: dict = None, headers: dict = None) -> Response:
```
Comment (ekansa) [inline isb_lib/utilities/requests_utilities.py]: Do you want to include a pause between retries? Or is that done elsewhere?

Comment (dannymandel) [reply isb_lib/utilities/requests_utilities.py]: It is not.  That is a good suggestion!

File: isb_lib/utilities/requests_utilities.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                response = rsession.get(url, params=params, headers=headers, timeout=self._timeout)
+                response.raise_for_status()
+                if self._success_func is not None:
+                    result = self._success_func(response)
+                    if not result:
+                        raise RequestException("Invalid response according to success function")
+                return response
+            except (RequestException, Timeout) as e:
+                L.warning(f"Request failed: {e}")
+                if self._sleep_sec > 0:
```
Comment (ekansa) [inline isb_lib/utilities/requests_utilities.py:34]: Great! That should be helpful to give a remote service time to recover from being temporarily too busy or give it time to restart if it failed.

Comment (dannymandel) [reply isb_lib/utilities/requests_utilities.py:34]: I set it to 60 to start with.

**Comment by dannymandel** [comment]
Going to merge this to push it out to henry.cyverse.org.

**Pull request merged**